play-json-tools/src/test/scala/com/evolutiongaming/playjsontools/DiscriminatedEitherFormatSpec.scala

```diff
@@ -0,0 +1,35 @@
+package com.evolutiongaming.playjsontools
+
+import com.evolutiongaming.playjsontools.PlayJsonHelper._
+import org.scalatest.funsuite.AnyFunSuite
+import org.scalatest.matchers.should.Matchers
+import play.api.libs.json._
+
+class DiscriminatedEitherFormatSpec extends AnyFunSuite with Matchers {
```

Comment:
please add test parsing ```{"L":"1", "R":1 }```

Reply:
sure, will do, but technically `test("eitherFormat should fail on extra keys")` already tests this

Reply:
this is rather interesting case,  when you have a valid input for reading from perspective of left side or right side, however not a valid in general :)